contracts/pcv/barnbridge/SmartYieldRedeemer.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface ISmartYield {
+    function sellTokens(
+      uint256 tokenAmount_,
+      uint256 minUnderlying_,
+      uint256 deadline_
+    ) external;
+
+    function transfer(address account, uint256 amount) external;
+}
+
+/// @title base class for a claiming BarnBridge Smar Yield tokens
+/// @author Fei Protocol
+contract SmartYieldRedeemer {
+    using SafeERC20 for IERC20;
+
+    address public immutable target;
+
+    IERC20 public immutable underlying;
+
+    constructor(
+        address _target,
+        IERC20 _underlying
+    ) { 
+        target = _target; 
+        underlying = _underlying;
+    }
+
+    /// @notice redeem BarnBridge SmartYield juniors
+    /// @param bbJunior smart yield token to redeem from
+    /// @param amount tokens to redeem
+    function redeem(ISmartYield bbJunior, uint256 amount) external { 
+        bbJunior.sellTokens(amount, 0, block.timestamp);
```

Comment:
somewhat risky to put a minimum amount out of 0, but given the amounts at stake, ok for me

Reply:
if we are selling into an amm, I think we risk getting front and back run so this function should have a minAmountOut param to limit damage from slippage